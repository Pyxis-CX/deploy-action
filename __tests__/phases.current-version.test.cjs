const fs = require('node:fs/promises')
const { describe, expect, test } = require('@jest/globals')
const core = require('@actions/core')

const { getCurrentVersions } = require('../src/phases/current-versions')

const client = {
  get: async () => {
    return Promise.resolve({
      readBody: async () => {
        return fs.readFile('./__tests__/sample-data/versions.json', 'utf-8')
      },
    })
  },
}

describe('current-version.js', () => {
  test('should return the current version', async () => {
    const runContext = {
      client,
      core,
      apiHostname: 'api.sandbox.arctesting2.arcpublishing.com',
    }
    const response = await getCurrentVersions(runContext)
    expect(response).toMatchSnapshot()
    expect(response).toHaveLength(2)
    expect(parseInt(response[1])).toBeGreaterThan(parseInt(response[0]))
  })
})
