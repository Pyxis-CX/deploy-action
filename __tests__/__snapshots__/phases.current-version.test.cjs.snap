// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`current-version.js should return the current version 1`] = `
[
  1,
  31,
]
`;
